routers/repo/issue_timetrack.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/auth"
+	"code.gitea.io/gitea/modules/context"
+)
+
+// AddTimeManually tracks time manually
+func AddTimeManually(c *context.Context, form auth.AddTimeManuallyForm) {
+	issueIndex := c.ParamsInt64("index")
+	issue, err := models.GetIssueByIndex(c.Repo.Repository.ID, issueIndex)
+	if err != nil {
+		c.Handle(http.StatusInternalServerError, "GetIssueByIndex", err)
+		return
+	}
+	url := issue.HTMLURL()
+
+	if c.HasError() {
+		c.Flash.Error(c.GetErrMsg())
+		c.Redirect(url)
+		return
+	}
+
+	h, err := parseTimeTrackingWithDuration(form.Hours, "h")
+	if err != nil {
+		c.Handle(http.StatusInternalServerError, "parseTimeTrackingWithDuration", err)
+		return
+	}
+
+	m, err := parseTimeTrackingWithDuration(form.Minutes, "m")
+	if err != nil {
+		c.Handle(http.StatusInternalServerError, "parseTimeTrackingWithDuration", err)
+		return
+	}
+
+	total := h + m
```

Comment:
I think you can simply do `total := form.Hours * time.Hour + form.Minutes + time.Minutes`

Reply:
I think the current solution is better because we have to parse form.Hours to Duration (instead of int64) to do this operation. We have currently a parse function (see parseTimeTrackingWithDuration) with the advantage to handle the duration units like minutes or hours.

Reply:
I don't think I'm following what you're saying. By using constants such as `time.Hour` and `time.Minutes`, we can handle units without parsing.

I'm not sure the current solution is optimal, because it is involving converting ints to strings (inside the `parseTimeTrackingWithDuration`), which are then parsed back into ints (by `time.ParseDuration`, note that the `time.Duration` type is simply defined as `int64`).